File: src/main/java/com/dndweapons/item/DndWeaponItem.java
package com.dndweapons.item;

import com.dndweapons.catalog.WeaponSpec;
import net.minecraft.item.Item;

/**
 * Generic melee weapon. Carries a back-reference to its spec so combat hooks
 * (Phase 3) can look up properties at runtime. Phase 1 does not implement
 * any combat hooks — this is just the item identity.
 */
public class DndWeaponItem extends Item {
    private final WeaponSpec spec;

    public DndWeaponItem(WeaponSpec spec, Settings settings) {
        super(settings);
        this.spec = spec;
    }

    public WeaponSpec spec() {
        return spec;
    }
}

File: src/main/kotlin/com/dndweapons/item/DndWeaponItem.kt
package com.dndweapons.item

import com.dndweapons.catalog.WeaponSpec
import com.dndweapons.compat.AttributeCompat
import net.minecraft.world.item.Item

open class DndWeaponItem(val spec: WeaponSpec, settings: Properties) : Item(settings) {

    init {
        AttributeCompat.storeFor(spec)
    }

    //? if <1.20.5 {
    /*
    override fun getAttributeModifiers(
        slot: net.minecraft.world.entity.EquipmentSlot,
    ): com.google.common.collect.Multimap<net.minecraft.world.entity.ai.attributes.Attribute, net.minecraft.world.entity.ai.attributes.AttributeModifier> {
        return AttributeCompat.modifiersFor(spec, slot)
    }
    */
    //?}
}
